Log mapping from WebContent pid to URLs

bnham opened the pull request:
#### b2803c68023a03aee80f2f063e44745f52587029
<pre>
Log mapping from WebContent pid to URLs
<a href="https://bugs.webkit.org/show_bug.cgi?id=270640">https://bugs.webkit.org/show_bug.cgi?id=270640</a>
<a href="https://rdar.apple.com/123269799">rdar://123269799</a>

Reviewed by Chris Dumez.

In some of our benchmarks, we need a way of figuring out which WebContent pid contains a particular
page URL. Extend the existing logProcessPoolState notification to log this.

Also fix a few other random issues:

- Move the taskInfo logging over to WebProcessProxy.
- Sometimes the registrable domain was empty, in that case log that the associated domain is unknown.

* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::logProcessPoolState):
(WebKit::description): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::operator&lt;&lt;):

Canonical link: <a href="https://commits.webkit.org/275923@main">https://commits.webkit.org/275923@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b217196ff685f0dd711e9521cebd2a740c2aef4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43300 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22330 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/45710 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/45934 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/39426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26078 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/19748 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/35799 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/43874 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/19359 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/37311 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/16786 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/16946 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1356 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/39488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/38688 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47479 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/18214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/14998 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/42591 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/19751 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/41247 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9631 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/19930 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/19382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
```diff
@@ -271,29 +273,28 @@ static AccessibilityPreferences accessibilityPreferences()
 
-        String domain = process->optionalRegistrableDomain() ? process->optionalRegistrableDomain()->string() : "unknown"_s;
-        RELEASE_LOG(Process, "WebProcessProxy %p - %" PUBLIC_LOG_STRING ", domain: %" PRIVATE_LOG_STRING, process.ptr(), stream.release().utf8().data(), domain.utf8().data());
+        RELEASE_LOG(Process, "WebProcessProxy %p - %" PUBLIC_LOG_STRING ", domain: %" PRIVATE_LOG_STRING ", pageURLs: %" SENSITIVE_LOG_STRING, process.ptr(), processDescription.release().utf8().data(), domainString.utf8().data(), pageURLs.release().utf8().data());
```
I see we're using `SENSITIVE_LOG_STRING` here so we're likely fine from a privacy standpoint.

bnham replied:
Correct, this isn't specific to benchmarks, but the requested use case was from a particular benchmark. And the logging has to be done using `%{sensitive}s`.

cdumez commented on Source/WebKit/UIProcess/WebProcessProxy.cpp:
```diff
@@ -2711,6 +2711,22 @@ TextStream& operator<<(TextStream& ts, const WebProcessProxy& process)
+
+    if (auto taskInfo = process.taskInfo()) {
+        ts << ", state: " << description(taskInfo->state);
+        ts << ", phys_footprint_mb: " << (taskInfo->physicalFootprint / 1048576.0) << " MB";
```
Please use `MB` from StdLibExtras.h:

`(taskInfo->physicalFootprint / static_cast<double>(MB))` or `(taskInfo->physicalFootprint / MB)` if we don't need floating point precision.

cdumez reviewed (approved):
LGTM with nit.

bnham committed:
Log mapping from WebContent pid to URLs
https://bugs.webkit.org/show_bug.cgi?id=270640
rdar://123269799

Reviewed by Chris Dumez.

In some of our benchmarks, we need a way of figuring out which WebContent pid contains a particular
page URL. Extend the existing logProcessPoolState notification to log this.

Also fix a few other random issues:

- Move the taskInfo logging over to WebProcessProxy.
- Sometimes the registrable domain was empty, in that case log that the associated domain is unknown.

* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::logProcessPoolState):
(WebKit::description): Deleted.
* Source/WebKit/UIProcess/WebProcessProxy.cpp:
(WebKit::operator<<):

Canonical link: https://commits.webkit.org/275923@main
